base/compiler/tfuncs.jl

```diff
@@ -500,6 +513,12 @@ function typeassert_tfunc(@nospecialize(v), @nospecialize(t))
     end
     return typeintersect(widenconst(v), t)
 end
+
+function typeassert_tfunc(@nospecialize(v), @nospecialize(t))
+    t = instanceof_tfunc(t)[1]
+    t === Any && return v
+    typeassert_instance(v, t)
```

Comment:
```suggestion
    return typeassert_instance(v, t)
```

I know a computer doesn't care about the different, but I treat it as a trivial comment for future human readers